test/src/unit/modules/Project.test.js

```diff
@@ -95,9 +108,82 @@ describe('Project.js', () => {
             json.should.not.containSubset(obj);
         });
     });
-    describe('getPort', () => {
+    describe('checkIfMetricsAvailable()', () => {
+        describe('Checks if metrics are available for Normal projects', () => {
+            it('Generic project with no language', async() => {
+                const project = new Project({ name: 'dummy' }, global.codewind.CODEWIND_WORKSPACE);
+                project.should.be.an('object');
+                const areMetricsAvailable = await project.checkIfMetricsAvailable();
+                areMetricsAvailable.should.be.false;
+            });
+            it('Node.js', async() => {
+                const project = new Project({ name: 'dummy', language: 'nodejs' }, global.codewind.CODEWIND_WORKSPACE);
+                project.should.be.an('object');
+                const packageJSONContents = {
+                    dependencies: {
+                        'appmetrics-dash': true,
+                    },
+                };
+                const packageJSONPath = path.join(project.projectPath(), 'package.json');
+                await fs.ensureFile(packageJSONPath);
+                await fs.writeJSON(packageJSONPath, packageJSONContents);
+                const areMetricsAvailable = await project.checkIfMetricsAvailable();
+                areMetricsAvailable.should.be.true;
+            });
+            it('Java', async() => {
+                const project = new Project({ name: 'dummy', language: 'java' }, global.codewind.CODEWIND_WORKSPACE);
+                project.should.be.an('object');
+                const pomXmlPath = path.join(project.projectPath(), 'pom.xml');
+                await fs.ensureFile(pomXmlPath);
+                await fs.writeFile(pomXmlPath, 'javametrics');
+                const areMetricsAvailable = await project.checkIfMetricsAvailable();
+                areMetricsAvailable.should.be.true;
+            });
+            it('Swift', async() => {
+                const project = new Project({ name: 'dummy', language: 'swift' }, global.codewind.CODEWIND_WORKSPACE);
```

Comment:
You could combine these 2 lines into a `createProjectAndCheckItsAnObject(projectParams)`
function, and call that at the start of each it block

Reply:
Would save 1/2 lines in lots of places